### DISCUSSION TITLE
Prisma Mongodb is AWESOME!

### DISCUSSION THREAD

**Discussion by masterbater** [prisma/prisma#12857] [Q&A] [answered]
For 3 days playing around with mongodb prisma, it was very enjoyable to use. It catches a lot of dumb errors,  intuitive suggestions and better DX than mongoose.  
What I like about prisma implementation

- Return data from the database query are also type check, example if one of your return field is  expected to be string but it returns null it will return an error.
- Almost one to one equivalent syntax like for relational database.  Database agnostic 😮
- Autocompletes your query, like wow it does really return what expected field or fields can be used for queries and even relations.
- Introspection is very useful for quickly to infer and create schema, it does only check 1000 documents and it will cause errors during queries but it helps you fixes those data types 
- Referential Actions

Prisma Team 
You really made a hard work with supporting mongodb especially with the 1:1 consistent syntax not like TypeORM and the Referential Actions support also adds data integrity and consistency. I hope you could continue to support and developed mongodb.

Once mongodb rust driver supports auto encryption prisma will be perfect for apps that focuses on end to end type safety, sensitive information and also forgot to say that prisma creates collection using transaction for that is crucial for data integrity like for payments. 
Projects like this makes  me also fall in love with typescript 
Congrats for GA 🎉🎉🎉🎉🎉

**Comment by sontiO** [comment, marked as answer]
I enjoyed it a lot in dev experience when fiddling with it.
WHen I tried to deploy my simple test on vercel I ran into some issues sadly.

ERROR	Error: ENOENT: no such file or directory, open '/var/task/schema.prisma' is my error on deployment.

**Comment by masterbater** [comment]
You can using query params but it is on by default. Its not an encryption
its an option to make mongodb manage all the connection and pools

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
